app/models/pubmed_source_record.rb

```diff
@@ -175,13 +176,8 @@ def source_as_hash(publication = Nokogiri::XML(source_data).xpath('//PubmedArtic
     year = extract_year_from_pubmed_record(publication)
     record_as_hash[:year] = year if year
 
-    month = extract_month_from_pubmed_record(publication)
-    day = extract_day_from_pubmed_record(publication)
-    if year && month && day
-      record_as_hash[:date] =  "#{year}-#{month.rjust(2, '0')}-#{day.rjust(2, '0')}"
-    elsif year && month
-      record_as_hash[:date] =  "#{year}-#{month.rjust(2, '0')}"
-    end
+    month_day = extract_month_day_from_pubmed_record(publication)
+    record_as_hash[:date] = "#{year}-#{month_day}" if year && month_day
 
```

Comment:
combine the day and month methods into one method, so this part of the code gets simpler